### PR TITLE
203 populate country field in user table when updating users ip address

### PR DESCRIPTION
**Pull request by SirL3X0N**
LocationController.php isn't used in my solution but the tutorials on the service were using are making use of it so I included it in the last commit

### CODE REVIEW

**Review by bytech-ca — CHANGES_REQUESTED** [review]
@SirL3X0N Items that require closer review:

- [ ] Per docs at https://packagist.org/packages/stevebauman/location, changes to app.php are not needed starting Laravel 5.5.
- [ ] Hard coded 89.151.39.138 IP address in LocationController, LoginListener, others?
- [ ] dd dump in LocationController looks out of place
- [ ] Changes to .env.example are needed... LOCATION_TESTING (others?) should be added

Here for any questions you might have.

File: app/Listeners/LoginListener.php [hunk 1]
```diff
@@ -25,10 +27,13 @@ public function __construct()
      */
     public function handle($event)
     {
+        $userIp = $request->ip();
+        $locationData = \Location::get($userIp);
```
Comment (bytech-ca) [inline app/Listeners/LoginListener.php:31]: Let's add a `use` statement instead the fully qualified class name `\Location` (see [this issue](https://github.com/eclectic-bytech/webcourses-ngx-v4/issues/192) for an example in the final comment's linked commit.) Also checkout [this 5 minute tutorial on name spaces](https://symfonycasts.com/screencast/php-namespaces/namespaces) that explains this implementation. 

The [official docs](https://packagist.org/packages/stevebauman/location) recommend putting an if statement around the Location::Get. This won't work inside `$event->user->update`. To get around that, we can set the country code to `XYZ` and then update it with a real one if found. This way there's a value for the update statement, even if the GeoID fails for any reason.

File: app/Listeners/LoginListener.php [hunk 2]
```diff
@@ -14,7 +17,6 @@ class LoginListener
      */
     public function __construct()
     {
-        //
```
Comment (bytech-ca) [inline app/Listeners/LoginListener.php:17]: An empty function might break tests, other things. Let's re-add the comment slashes.

File: app/Listeners/LoginListener.php [hunk 3]
```diff
@@ -4,6 +4,9 @@
 
 use Illuminate\Contracts\Queue\ShouldQueue;
 use Illuminate\Queue\InteractsWithQueue;
+use App\Http\Controllers\LocationController;
+use Illuminate\Support\Facades\Route;
+use Illuminate\Http\Request;
```
Comment (bytech-ca) [inline app/Listeners/LoginListener.php:10]: Route and Request imports might not be needed after making changes listed below.

**Review by bytech-ca — CHANGES_REQUESTED** [review]

**Comment by SirL3X0N** [comment]
@bytech-ca  ready for review

**Pull request merged**